aips/create_dip.py

```diff
@@ -0,0 +1,224 @@
+    LOGGER.info('Extracting AIP')
+    aip_name = extract_aip(aip_file, aip_uuid, tmp_dir)
+
+    if not aip_name:
+        LOGGER.error('Unable to extract AIP')
+        return
+
+    aip_dir = os.path.join(tmp_dir, aip_name)
+
+    LOGGER.info('Creating DIP')
+    dip_dir = create_dip(aip_dir, aip_uuid)
+
+    if not dip_dir:
+        LOGGER.error('Unable to create DIP')
+        return
+
+    LOGGER.info('DIP created in: %s', dip_dir)
+
+
+def download_aip(ss_url, ss_user, ss_api_key, aip_uuid, tmp_dir):
+    """Download the AIP from Storage Service"""
+    aip_url = '{}/api/v2/file/{}/download/'.format(ss_url, aip_uuid)
+    params = { 'username': ss_user, 'api_key': ss_api_key }
+    LOGGER.debug('SS AIP URL: %s', aip_url)
+
+    response = requests.get(aip_url, params, stream=True)
+    if response.status_code == 200:
+        try:
+            aip_filename = re.findall(
+                'filename="(.+)"',
+                response.headers['content-disposition'])[0]
+        except KeyError:
+            # Assuming .7z format if content-disposition is missing
+            LOGGER.warning('Response headers is missing content-disposition')
+            aip_filename = 'Untitled-{}.7z'.format(aip_uuid)
+
+        LOGGER.debug('AIP filename: %s', aip_filename)
+
+        aip_path = os.path.join(tmp_dir, aip_filename)
+        with open(aip_path, 'wb') as file_:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    file_.write(chunk)
+
+        return aip_path
+
+
+def extract_aip(aip_file, aip_uuid, tmp_dir):
+    """Extract a downloaded AIP to a folder. Accepted formats: '.tar', '.7z'"""
+
+    # tar archives, including those using gzip or bz2 compression
+    if tarfile.is_tarfile(aip_file):
+        try:
+            tar = tarfile.open(aip_file)
+
+            # Get top-level folders from tar file
+            dirs = []
+            for tarinfo in tar:
+                if '/' not in tarinfo.name and tarinfo.isdir():
+                    dirs.append(tarinfo.name)
+
+            if len(dirs) is not 1:
+                LOGGER.warning('AIP has none or more than one folder')
+                return
+
+            LOGGER.debug('AIP dir: %s', dirs[0])
+            tar.extractall(tmp_dir)
+            tar.close()
+
+            return dirs[0]
+        except tarfile.TarError as err:
+            LOGGER.warning('Tarfile error: {}. Trying with /bin/tar'.format(err))
+
+    # 7z, failed tar and other archives based on file last extension
+    ext = aip_file.split('.')[-1]
+    command = {
+        'tar': ['tar', 'xvf', aip_file, '-C', tmp_dir],
+        'bz2': ['tar', 'xvjf', aip_file, '-C', tmp_dir],
+        'gz': ['tar', 'xvzf', aip_file, '-C', tmp_dir],
+        '7z': ['7z', 'x', '-bd', '-y', '-o{0}'.format(tmp_dir), aip_file]
+    }.get(ext, ['unar', '-force-overwrite', '-o', tmp_dir, aip_file])
+
+    LOGGER.debug('Extract command: %s', command)
+    if subprocess.call(command) is not 0:
```

Comment:
`subprocess.check_call` is probably a better choice because it raises an exception (more idiomatic in Python) and the exception has an `output` attribute that you could use to show to the user what happened with the underlying tool.